drivers/gpio/gpio_mchp_xec.c

```diff
@@ -215,7 +219,13 @@ static int gpio_xec_pin_interrupt_configure(const struct device *dev,
 	 */
 	current_pcr1 = config->pcr1_base + pin;
 	*current_pcr1 = (*current_pcr1 & ~mask) | pcr1;
-	__DMB(); /* insure write completes */
+	/* delay for HW to synchronize after it ungates its clock */
+	for (int i = 0; i < XEC_GPIO_EDGE_DLY_COUNT; i++) {
+		mask = *current_pcr1;
+		mask <<= 1;
+	}
```

Comment:
An easier way to write the loop would be
```
	for (int i = 0; i < XEC_GPIO_EDGE_DLY_COUNT; i++) {
		(void)*current_pcr1;
	}
```
In this case there is no need for `sys_write32(mask, XEC_GPIO_DLY_ADDR);` The `XEC_GPIO_DLY_ADDR` define can be removed.

`current_pcr1` is defined as pointer to volatile so compiler will not optimize it away.

Reply:
@mnkp Changes done